examples/master.js

```diff
@@ -170,36 +200,37 @@ async function startMaster(localView, remoteView, formValues, onStatsReport, onR
         }
 
         master.signalingClient.on('open', async () => {
+            const masterRunId = ++master.runId;
+            master.websocketOpened = true;
             console.log('[MASTER] Connected to signaling service');
-            if (formValues.ingestMedia && master.streamARN) {
-                try {
-                    console.log('[MASTER] Joining storage session...');
-                    await master.storageClient
-                        .joinStorageSession({
-                            channelArn: master.channelARN,
-                        })
-                        .promise();
-
-                    console.log('[MASTER] Joined storage session. Media is being recorded to', master.streamARN);
-                } catch (e) {
-                    console.error('[MASTER] Error joining storage session', e);
+            if (master.streamARN) {
+                if (formValues.ingestMedia) {
+                    await connectToMediaServer(masterRunId);
+                } else {
+                    console.log('[MASTER] Waiting for media ingestion and storage viewer to join...');
                 }
+            } else {
+                console.log('[MASTER] Media ingestion and storage is not enabled for this channel. Waiting for peers to join...');
             }
         });
 
         master.signalingClient.on('sdpOffer', async (offer, remoteClientId) => {
             printSignalingLog('[MASTER] Received SDP offer from client', remoteClientId);
+            master.sdpOfferReceived = true;
```

Comment:
Does this cover the case where we might receive multiple offers for the same connection?

Reply:
Yes